### PR TITLE
CollapsableSection: Improves keyboard navigation and screen-reader support

### PR DESCRIPTION
**Pull request by kaydelaney**
**What this PR does / why we need it**:
Adds keyboard support and appropriate aria attributes to the `CollapsableSection` component. Changes `SearchResults`/`SectionHeader` to make use of `CollapsableSection`.

### CODE REVIEW

**Review by ashharrison90 — APPROVED** [review]
lgtm! 🙌

**Comment by kaydelaney** [comment]
Sp, test-a11y-frontend reports the following:

```
Errors in http://end-to-end-tests-server:3001/d/O6f11TZWk/panel-tests-bar-gauge?orgId=1&editview=settings:


 • Elements must only use allowed ARIA attributes

   (https://dequeuniversity.com/rules/axe/4.3/aria-allowed-attr?application=axeAPI)


   (#collapse-content-2)
```

Double-checked this with axe devtools which says
```ARIA attribute: aria-labelledby is not well supported. Use a different role attribute or element.```

The trouble is, I don't think that's true? It's listed as part of [wai-aria](https://www.w3.org/TR/wai-aria-1.1/#aria-labelledby), and according to [a11ysupport.io](https://a11ysupport.io/tech/aria/aria-labelledby_attribute), it's supported by almost all major screen readers. Not sure what's going on to be honest!

File: packages/grafana-ui/src/components/Collapse/CollapsableSection.tsx [hunk 1]
```diff
@@ -1,55 +1,117 @@
...
+          {label}
+        </div>
       </div>
-      {open && <div className={styles.content}>{children}</div>}
-    </div>
+      {open && (
+        <div
+          id={`collapse-content-${id}`}
+          className={cx(styles.content, contentClassName)}
+          aria-labelledby={buttonLabelId}
```
[the remark this answers is not in the document]
Comment (kaydelaney) [reply packages/grafana-ui/src/components/Collapse/CollapsableSection.tsx:70]: Nah, see on line 45:
```ts
const buttonLabelId = labelId ?? `collapse-label-${id}`;
```

and then on line 62 (admittedly, could probably use a variable for `collapse-label-${id}`)
```ts
  <div className={styles.label} id={`collapse-label-${id}`}>
```

So by default it points to that element, with the option of having the label pointed elsewhere by passing `labelId` to `CollapsableSection`.

[the remark this answers is not in the document]
Comment (ashharrison90) [reply packages/grafana-ui/src/components/Collapse/CollapsableSection.tsx:70]: ahhhhhhh gotcha, completely missed that 👍

File: packages/grafana-ui/src/components/Collapse/CollapsableSection.tsx [hunk 2]
```diff
@@ -1,55 +1,117 @@
...
   const tooltip = `Click to ${open ? 'collapse' : 'expand'}`;
-  const onClick = () => {
+  const onClick = (e: React.MouseEvent) => {
+    e.preventDefault();
+    e.stopPropagation();
+
     onToggle?.(!open);
     toggleOpen(!open);
   };
+  const { current: id } = useRef(uniqueId());
```
Comment (joshhunt) [inline packages/grafana-ui/src/components/Collapse/CollapsableSection.tsx:41]: Do we want to use `useId` instead here? https://react-spectrum.adobe.com/react-aria/useId.html

**Review by joshhunt — APPROVED** [review]

**Comment by joshhunt** [comment]
oh, that meant to be just a comment, not necessarily an approval 👍🏻  fwiw

**Pull request merged**